packages/effect/src/Effect.ts

```diff
@@ -594,18 +594,23 @@ export declare namespace All {
    */
   export type ExtractMode<A> = [A] extends [{ mode: infer M }] ? M : "default"
 
+  /**
+   * @since 3.7.0
+   */
+  export type Options = {
+    readonly concurrency?: Concurrency | undefined
+    readonly batching?: boolean | "inherit" | undefined
+    readonly discard?: boolean | undefined
+    readonly mode?: "default" | "validate" | "either" | undefined
+    readonly concurrentFinalizers?: boolean | undefined
+  }
+
   /**
    * @since 2.0.0
    */
   export type Return<
     Arg extends Iterable<EffectAny> | Record<string, EffectAny>,
-    O extends {
```

Comment:
I think this should be reverted, and then remove the `discard` option.

Reply:
we also need the "concurrentFinalizers" option